src/main/java/com/example/animalshelter/controller/AnimalController.java

```diff
@@ -30,7 +30,7 @@ public String addAnimal(@RequestBody Animal animal) {
     }
 
     @GetMapping("/{id}")
-    public Animal one(@PathVariable Long id) {
+    public Animal one(@PathVariable int id) {
         return animalService.findOne(id);
```

Comment:
Sorry for misleading with id type - we should stay with `Long` IMO. Please correct this.